modules/cmd.js

```diff
@@ -7,60 +7,61 @@ const getStateSymbol = Symbol('getState');
 
 
 const cmdTypes = {
-  PROMISE: 'PROMISE',
-  CALL: 'CALL',
-  CALLBACK: 'CALLBACK',
-  CONSTANT: 'CONSTANT',
-  ARBITRARY: 'ARBITRARY',
+  RUN: 'RUN',
+  ACTION: 'ACTION',
+  //CALLBACK: 'CALLBACK',
   BATCH: 'BATCH',
   MAP: 'MAP',
   NONE: 'NONE',
   SEQUENCE: 'SEQUENCE'
 }
 
-
 export const isCmd = (object) => {
   return object ? object[isCmdSymbol] : false
 }
 
-function getMappedCmdArgs(args, dispatch, getState){
+function getMappedCmdArgs(args = [], dispatch, getState){
   return args.map(arg => {
     if (arg === dispatchSymbol) return dispatch
     else if (arg === getStateSymbol) return getState
     else return arg
   })
 }
 
-export const cmdToPromise = (cmd, dispatch, getState) => {
-  switch (cmd.type) {
-    case cmdTypes.PROMISE:
-      return cmd.promiseFactory(...getMappedCmdArgs(cmd.args, dispatch, getState))
-        .then(cmd.successActionCreator)
-        .catch(cmd.failureActionCreator)
-        .then((action) => {
-          return [action]
-        })
+function handleRunCmd(cmd, dispatch, getState){
+  let onSuccess = cmd.successActionCreator || (() => {}),
+      onFail = cmd.failActionCreator || (() => {})
 
-    case cmdTypes.CALL:
-      const result = cmd.resultFactory(...getMappedCmdArgs(cmd.args, dispatch, getState))
-      return Promise.resolve([cmd.actionCreator(result)])
+  try{
+    let result = cmd.func(...getMappedCmdArgs(cmd.args, dispatch, getState))
 
-    case cmdTypes.CALLBACK:
-      return promisify(cmd.nodeStyleFunction)(...getMappedCmdArgs(cmd.args, dispatch, getState))
-        .then(cmd.successActionCreator)
-        .catch(cmd.failureActionCreator)
-        .then((action) => [action])
+    if (isPromiseLike(result) && !cmd.forceSync){
+      return result.then(onSuccess, onFail).then(action => {
+        return action ? [action] : null;
+      })
```

Comment:
yea it just ended up being a long line so i split it up.

i think null is preferable because things like batch cmds (or the top level cmd runner), filter out falsey things so they only dispatch actions and not null/undefined. an empty would be truthy so it wouldn't get filtered